master/internal/api_config_policies.go

```diff
@@ -67,10 +67,36 @@ func (*apiServer) GetWorkspaceConfigPolicies(
 	ctx context.Context, req *apiv1.GetWorkspaceConfigPoliciesRequest,
 ) (*apiv1.GetWorkspaceConfigPoliciesResponse, error) {
 	if !configpolicy.ValidWorkloadType(req.WorkloadType) {
-		return nil, fmt.Errorf("invalid workload type: %s", req.WorkloadType)
+		errMessage := fmt.Sprintf("invalid workload type: %s.", req.WorkloadType)
+		if len(req.WorkloadType) == 0 {
+			errMessage = noWorkloadErr
+		}
+		return nil, status.Errorf(codes.InvalidArgument, errMessage)
 	}
-	data, err := stubData()
-	return &apiv1.GetWorkspaceConfigPoliciesResponse{ConfigPolicies: data}, err
+
+	resp := apiv1.GetWorkspaceConfigPoliciesResponse{}
+	configPolicies, err := configpolicy.GetTaskConfigPolicies(
+		ctx, ptrs.Ptr(int(req.WorkspaceId)), req.WorkloadType)
+	if err != nil {
+		return nil, err
+	}
+	policyMap := map[string]interface{}{}
```

Comment:
This method of converting the configs & constraints to something marshal-able definitely works. It feels a little complicated though.

What about concatenating the strings and doing something like what's in the `stubData()` function? 
```
	var yamlMap map[string]interface{}
	if err := yaml.Unmarshal([]byte(yamlString), &yamlMap); err != nil {
		return nil, fmt.Errorf("unable to unmarshal yaml: %w", err)
	}

	return configpolicy.MarshalConfigPolicy(yamlMap), nil
```

Reply:
discussed with Kristine offline but concatenating strings is pretty complicated if we do not save the "invariant_config" layer around the data while saving the json in the db so going to keep the logic as is